### PR TITLE
2D Textured Quad w/PNGs via stb_image

### PR DESCRIPTION
**Pull request by OniEnzeru**
Created a simple and heavily commented example to help new users understand how to render a 2D textured Quad using OpenGldc and PNG textures ( for ease of entry ).  Loading is done via stb_image to further lower the bar to entry.

Has controller input:
start: close program
Y/ B: Scale up /down
LR Trigger: Rotate CCW / CW

output

### CODE REVIEW

**Comment by QuzarDC** [comment]
Is there a reason to avoid using just libpng that we have in kos-ports rather than the stb_image (other than it just being what you are already using)? If it's just for the png loading it would be better to do that. 

If not it's effectively a port living in examples that wouldn't benefit from easy updates. It would make sense though to perhaps figure out how to have it as part of kos-ports if it works right out of the box. Though it's not the typical 'build a library' thing, it should be able to be set up as something to pull down from git. Otherwise seems like a good simple example.

**Comment by OniEnzeru** [comment]
It was for easy png ( but can be used for most modern image formats) loading as it removed most of the boiler plate around loading in the header, reading the formatting etc.  It allowed the example to come in at a much higher level by just simplifying it, ultimately, to a single line of code.

I was just trying to make it as simple as possible for new programmers. But having stb_image as a KOS PORT would be awesome if that's something that could happen or how it should be handled.

**Comment by OniEnzeru** [comment]
Spoke with @gyrovorbis about this and we think putting this on hold and working towards making stb_image a KOS port would be very beneficial.  As a single header library it can handle many image formats such as:

 Image formats                                       Extensions, features
    Sean Barrett (jpeg, png, bmp)           Jetro Lauha (stbi_info)
    Nicolas Schulz (hdr, psd)                   Martin "SpartanJ" Golini (stbi_info)
    Jonathan Dummer (tga)                     James "moose2000" Brown (iPhone PNG)
    Jean-Marc Lienher (gif)                      Ben "Disch" Wenger (io callbacks)
    Tom Seddon (pic)                               Omar Cornut (1/2/4-bit PNG)
    Thatcher Ulrich (psd)                          Nicolas Guillemot (vertical flip)
    Ken Miller (pgm, ppm)                        Richard Mitton (16-bit PSD)
    github:urraka (animated gif)                Junggon Kim (PNM comments)
    Christopher Forseth (animated gif)     Daniel Gibson (16-bit TGA)
                                                                socks-the-fox (16-bit PNG)
                                                                Jeremy Sawicki (handle all ImageNet JPGs)
                                                                
As a simple, single header library.  This ultimately would reduce what's needed for to load things from libraries like libPNG to a single line of code, eliminating the need for all the boiler plate.

**Comment by OniEnzeru** [comment]
Okay now that my STB_Image KOS-Port has been merged, I updated the example to take advantage of it.

File: examples/dreamcast/gldc/2D_Textured_Quad_Example-GLDC/Makefile [hunk 1]
```diff
@@ -0,0 +1,38 @@
+#
+# 2D_Textured_Quad_PNG Makefile
+#
+# Author: Jason Rost (OniEnzeru)
+# Date:	  1/22/2024
+#
```
Comment (ljsebald) [inline examples/dreamcast/gldc/2D_Textured_Quad_Example-GLDC/Makefile:6]:
```suggestion
#
# KallistiOS ##version##
#
# examples/dreamcast/gldc/2D_Textured_Quad_Example-GLDC/Makefile
# Copyright (C) 2024 Jason Rost (OniEnzeru)
#
```

**Review by ljsebald — CHANGES_REQUESTED** [review]
A bunch of style changes to make things match more closely to the rest of the examples...

**Review by ljsebald — APPROVED** [review]

**Pull request merged**